File: mobile_jamaah/lib/features/hotel/domain/hotel.dart
class Hotel {
  const Hotel({
    required this.id,
    required this.name,
    required this.city,
    this.address,
    this.latitude,
    this.longitude,
  });

  final int id;
  final String name;
  final String city;
  final String? address;
  final double? latitude;
  final double? longitude;

  factory Hotel.fromJson(Map<String, dynamic> json) => Hotel(
    id: json['id'] as int,
    name: json['name'].toString(),
    city: json['city'].toString(),
    address: json['address']?.toString(),
    latitude: (json['latitude'] as num?)?.toDouble(),
    longitude: (json['longitude'] as num?)?.toDouble(),
  );
}

File: app/Models/Hotel.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class Hotel extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'branch_id',
        'name',
        'city',
        'address',
        'latitude',
        'longitude',
        'geofence_radius_meters',
    ];

    protected function casts(): array
    {
        return [
            'latitude' => 'decimal:7',
            'longitude' => 'decimal:7',
            'geofence_radius_meters' => 'integer',
        ];
    }

    public function branch(): BelongsTo
    {
        return $this->belongsTo(Branch::class);
    }

    public function departures(): BelongsToMany
    {
        return $this->belongsToMany(Departure::class, 'departure_hotel')
            ->withPivot(['id', 'check_in_at', 'check_out_at', 'sequence'])
            ->withTimestamps();
    }
}
